DataAccess/Repository/OrderRepository.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using DataAccess.Data;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+
+namespace DataAccess.Repository
+{
+    public class OrderRepository : IOrderRepository
+    {
+        private BarDataBase _barDataBase;
+
+        public OrderRepository()
+        {
+            _barDataBase = new BarDataBase();
+        }
+
+        public async Task<Order> Get(int id)
+        {
+            Order order = await _barDataBase.Orders.FindAsync(id);
+            if(order == null)
+            {
+                throw new ArgumentException($"No order with {id} was found");
+            }
+            else
+            {
+                return order;
+            }
+        }
+
+        public async Task<int> Upsert(Order order)
+        {
+            bool ord = _barDataBase.Orders.Contains(order);
```

Comment:
I am not EF proficient user here.
But if this method goes to database it should be async. Why not use "Get" method from above? If it does not go to database - then i dont understand how it works. 

I think something wrong is going on here

Reply:
have you tested this method? Does it work as intended?
I still think that all operations with external dependencies(such as SQL DB) should be done asynchronously